In [14]:
import pandas as pd
import numpy as np

In [22]:
chunk_iter = pd.read_csv("data/MICRODADOS_ENEM_2019.csv", encoding = 'latin1', sep = ";", chunksize = 1000000)

In [23]:
total_rows = 0
chunks_filtered = []
for chunk in chunk_iter:
    total_rows += chunk.shape[0]
    chunks_filtered.append(chunk[chunk['SG_UF_RESIDENCIA'] == 'SP'])
df_sp = pd.concat(chunks_filtered)

In [24]:
print(f"Total rows: {total_rows}")
print(f"Total columns: {df_sp.shape[1]}")
print()
print("Keeping only the rows with data from the state of São Paulo")
print(f"Remaining rows: {df_sp.shape[0]}")
print("Removing columns with information that will not be used:")
print("Columns with information about each one of the question from the test")
print("TX_RESPOSTAS_CN   TX_RESPOSTAS_CH   TX_RESPOSTAS_LC   TX_RESPOSTAS_MT  \nTX_GABARITO_CN   TX_GABARITO_CH   TX_GABARITO_LC   TX_GABARITO_MT")
df_sp = df_sp.drop(columns = ['TX_RESPOSTAS_CN', 'TX_RESPOSTAS_CH','TX_RESPOSTAS_LC','TX_RESPOSTAS_MT',
                              'TX_GABARITO_CN','TX_GABARITO_CH','TX_GABARITO_LC','TX_GABARITO_MT'])
print(f"Remaing columns: {df_sp.shape[1]}")

Total rows: 5095270
Total columns: 136

Keeping only the rows with data from the state of São Paulo
Remaining rows: 813772
Removing columns with information that will not be used:
Columns with information about each one of the question from the test
TX_RESPOSTAS_CN   TX_RESPOSTAS_CH   TX_RESPOSTAS_LC   TX_RESPOSTAS_MT  
TX_GABARITO_CN   TX_GABARITO_CH   TX_GABARITO_LC   TX_GABARITO_MT
Remaing columns: 128


In [19]:
df_sp.to_csv("data/MICRODADOS_ENEM_SP_2019.csv", index = False)

In [2]:
df = pd.read_csv('data/MICRODADOS_ENEM_SP_2019.csv')

In [3]:
df.shape

(712032, 128)

In [7]:
columns_with_nans = list(df.isna().sum() > 0)
df.columns[columns_with_nans]

Index(['CO_MUNICIPIO_NASCIMENTO', 'NO_MUNICIPIO_NASCIMENTO',
       'CO_UF_NASCIMENTO', 'SG_UF_NASCIMENTO', 'TP_ENSINO', 'CO_ESCOLA',
       'CO_MUNICIPIO_ESC', 'NO_MUNICIPIO_ESC', 'CO_UF_ESC', 'SG_UF_ESC',
       'TP_DEPENDENCIA_ADM_ESC', 'TP_LOCALIZACAO_ESC', 'TP_SIT_FUNC_ESC',
       'CO_PROVA_CN', 'CO_PROVA_CH', 'CO_PROVA_LC', 'CO_PROVA_MT',
       'NU_NOTA_CN', 'NU_NOTA_CH', 'NU_NOTA_LC', 'NU_NOTA_MT',
       'TP_STATUS_REDACAO', 'NU_NOTA_COMP1', 'NU_NOTA_COMP2', 'NU_NOTA_COMP3',
       'NU_NOTA_COMP4', 'NU_NOTA_COMP5', 'NU_NOTA_REDACAO'],
      dtype='object')

Colunas com valores NAN que podem ser removidas: ['CO_MUNICIPIO_NASCIMENTO', 'CO_UF_NASCIMENTO', 'CO_MUNICIPIO_ESC', 'CO_UF_ESC', 'CO_PROVA_CN', 'CO_PROVA_CH', 'CO_PROVA_LC', 'CO_PROVA_MT'].

In [10]:
columns_not_useful =  ['CO_MUNICIPIO_NASCIMENTO', 'CO_UF_NASCIMENTO', 'CO_MUNICIPIO_ESC', 'CO_UF_ESC', 
                       'CO_PROVA_CN', 'CO_PROVA_CH', 'CO_PROVA_LC', 'CO_PROVA_MT']
df = df.drop(columns = columns_not_useful)

In [16]:
print('Colunas com NANs e a taxa de NANs')
nan_tx = list(df.isna().mean())
for i, col in enumerate(df.columns):
    if nan_tx[i] > 0:
        print(col + ': ' + str(np.round(nan_tx[i], 3)))

Colunas com NANs e a taxa de NANs
NO_MUNICIPIO_NASCIMENTO: 0.024
SG_UF_NASCIMENTO: 0.024
TP_ENSINO: 0.466
CO_ESCOLA: 0.74
NO_MUNICIPIO_ESC: 0.74
SG_UF_ESC: 0.74
TP_DEPENDENCIA_ADM_ESC: 0.74
TP_LOCALIZACAO_ESC: 0.74
TP_SIT_FUNC_ESC: 0.74
NU_NOTA_CN: 0.289
NU_NOTA_CH: 0.242
NU_NOTA_LC: 0.242
NU_NOTA_MT: 0.289
TP_STATUS_REDACAO: 0.242
NU_NOTA_COMP1: 0.242
NU_NOTA_COMP2: 0.242
NU_NOTA_COMP3: 0.242
NU_NOTA_COMP4: 0.242
NU_NOTA_COMP5: 0.242
NU_NOTA_REDACAO: 0.242


Cerca de 74% dos participantes não responderam informações sobre a escola em que estudam, isso é uma taxa bastante alta de informações que seriam úteis.

Além disso, a nota do aluno não está presenta para cerca de 24% dos alunos, informação que seria útil para realizar o treinamento de modelos.